publicAPI/crest_utils.py

```diff
@@ -294,12 +295,20 @@ def fetch_esi_endpoint(
         'User-Agent': config.get('GLOBAL', 'useragent')
     }
 
-    # no try-except, catch in caller
-    # done to make logging path easier
-    req = requests.get(
-        esi_url,
-        headers=headers
-    )
+    while True:
```

Comment:
new code requires coverage.  Coverage is:
-- tests that can be run in the unit tests
-- `logger.error()` that alerts humans of issues

This fails both as testable (need a way to magic in 502 errors) and does not alert people if recovery fails